static/app/views/settings/components/dataSecrecy/index.tsx

```diff
@@ -123,10 +145,20 @@ export default function DataSecrecy() {
     help: t(
       'Open a temporary time window for Sentry employees to access your organization'
     ),
-    disabled: allowAccess && !organization.access.includes('org:write'),
-    value: allowAccess ? '' : allowDateFormData,
+    // disable the field if the user has allowed access or if the user does not have org:write access
+    disabled: allowAccess || !organization.access.includes('org:write'),
```

Comment:
this was a bug, we should disable the date input if data secrecy is disabled OR if the user doesn't have enough permissions